Refactoring and randomization

jk370 opened the pull request:
Merging all recent changes described below into master branch. Game is currently split into four quadrants, with one room in each and a gold in the middle of that room.

All rooms and doors are randomized, with out of bounds errors fixed. No current problems found thus far.

Tidied up a lot of dead code.

theaaron123 committed:
added dungeon model and controller, prints empty dungeon

theaaron123 committed:
we added dungeon view, brittle implementation of movement needs refactor

theaaron123 committed:
extracted adding room to method - added TODO.

Adlington95 committed:
some changes as we discussed yesterday to make dungeon scaling and player spawn less hard coded and more flexible.

Adlington95 committed:
Fixed dungeon size to 20 by 20.

Adlington95 committed:
Set the colour scheme to rogue-like. Ready for lab demonstration.

jk370 committed:
Created constructor for Gold class and added GOLD_SYMBOL.
Changed PLAYER_SYMBOL to static and added getter.
Instantiated Player in DungeonController and removed playerPos.
Dungeon view now edits Player x and y through dungeon controller.

jk370 committed:
Added default close operation so game exits when window is closed.

Adlington95 committed:
I have added the following:
- Collision detection with unlimited rooms.
- Door creation and removal from the collision detection.
- Door entrance is logically positioned based on the position of the room in the dungeon.

jk370 committed:
Merge branch 'dungeon-initial' of https://github.com/theaaron123/dungeonsgame into instantiate-player-and-gold

# Conflicts:
#	src/main/java/DungeonController.java
#	src/main/java/DungeonView.java

Merged dungeon room update with instantiation of player
Fixed playerX and playerY being the wrong way round for movement
Minor changes to remove redundant code

jk370 committed:
Merge branch 'dungeon-initial' of https://github.com/theaaron123/dungeonsgame into instantiate-player-and-gold

# Conflicts:
#	src/main/java/DungeonController.java
#	src/main/java/DungeonView.java

Merged dungeon room update with instantiation of player
Fixed playerX and playerY being the wrong way round for movement
Minor changes to remove redundant code
Instantiated Gold with an arbritrary position

jk370 committed:
Merge branch 'dungeon-initial' of https://github.com/theaaron123/dungeonsgame into instantiate-player-and-gold

# Conflicts:
#	src/main/java/DungeonController.java
#	src/main/java/DungeonView.java

Merged dungeon room update with instantiation of player
Fixed playerX and playerY being the wrong way round for movement
Minor changes to remove redundant code
Instantiated Gold with an arbritrary position

Adlington95 committed:
can pick up the gold.
gold is in the middle of the room.
rooms are random.
exit position added.
passages added.

others:
dungeon size x 2
split into four large rooms.

Adlington95 committed:
marc/ jordan
rooms one per room
added passages to each quadrant
others..

Adlington95 committed:
marc/ jordan
Rooms now do not overlap with the quadrant edges.
Gold is now spawned in each room and can be removed from the game.
The player score is impacted from the multiple gold instances.

jk370 committed:
Fixed array out of bound errors
Made exit location random
Made room doors random
A great deal of refactoring and code clean up

jk370 committed:
Removed random println messages
change win amount from fixed value to depend on amount of gold
changed some access modifiers to private

theaaron123 commented on src/main/java/DungeonController.java:
```diff
@@ -0,0 +1,242 @@
+        int exitLocation;
+        int wall;
+        Random rand = new Random();
+        wall = rand.nextInt(4); //Random integer between 1 and 4
```
nextInt(int) is inclusive of zero, is this desired behaviour

theaaron123 replied:
Assuming not as you switch on 0 - 3

theaaron123 commented on src/main/java/DungeonController.java:
```diff
@@ -0,0 +1,242 @@
+        int startHeight;
+        int startWidth;
+
+        for (int r = 1; r <= rooms; r++) { //1 - top left, 2 - top right, 3 - bottom left, 4 - bottom right
```
is 'r' a room if so rename

theaaron123 commented on src/main/java/DungeonController.java:
```diff
@@ -0,0 +1,242 @@
+
+            switch (r) {
+                case 1:
+                    startHeight = rand.nextInt((dungeon.getHeight() / 2 - 6) - 2) + 2;
```
can you declare ints 2 and 6 to make more readable and only need changing in one place - assuming they are max and min

theaaron123 commented on src/main/java/DungeonController.java:
```diff
@@ -0,0 +1,242 @@
+        }
+    }
+
+    private void addGold(int room, int sh, int sw) {
```
sh and sw aren't intuitive names what are passing

theaaron123 commented on src/main/java/DungeonController.java:
```diff
@@ -0,0 +1,242 @@
+        Random rand = new Random();
+        int wall = rand.nextInt(4);
+
+        switch (wall) {
```
Used this case structure twice now, consider enum rather than comments of what they are

theaaron123 commented on src/main/java/DungeonController.java:
```diff
@@ -0,0 +1,242 @@
+    public void checkExit() {
+        if (gridBounds[player.getPlayerY()][player.getPlayerX()] == 5) {
+            if (player.getGold() >= gameWinAmount) {
+                System.out.println("Completed");
```
Is this debug, if not the user probably wont see this unless they launch from a shell

theaaron123 commented on src/main/java/DungeonView.java:
```diff
@@ -0,0 +1,138 @@
+
+    public DungeonView() {
+        final JFrame frame = new JFrame();
+        JPanel panel = new JPanel();
```
I know I named these with Marc but they need descriptive names before merge

theaaron123 commented on src/main/java/DungeonView.java:
```diff
@@ -0,0 +1,138 @@
+
+        frame.setVisible(true);
+
+        dg = new DungeonController();
```
dungeonController over dg

theaaron123 reviewed (changes requested):
Couple of changed before merge

jk370 committed:
A range of variables renamed to something more descriptive
Player turn added in preparation for bot
Game resets upon completion
sets gameWinAmount using number of rooms in dungeon
Prints array for debugging after full initialisation
checkExit now returns boolean to prevent out of bounds error

jk370 commented:
All requested changes actioned apart from enum as I am not sure of the best way to implement this.

jk370 committed:
removed debugging println statements
